lesson4

Relaxiy opened the pull request.

Relaxiy commented:
Я немного не понял как там 6 задание делать, сделал как смог

damalikov commented on src/lesson4/Task5/NotifierHandler.java:
```diff
@@ -0,0 +1,20 @@
+package lesson4.Task5;
+
+public class NotifierHandler {
+    public static void addNotifier(Notifier notifier){
```
Педполагалось, что addNotifier будет добавлять Notifier к какому-нибудь массиву всех уведомителей. 
Тогда метод notifyAll, будет брать список уведомителей в самом классе.

damalikov commented on src/lesson4/Task6/Person.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    public static Person worksLonger(Person[] person){
+        Person[] lectors = new Lector[person.length];
```
На самом деле можно сделать массив Lector[] lector.
Далее ты все равно кастишь Person к Lector, то есть проблем не будет. И после тебе не придется делать сново преобразование к Lector когда ты достаешь из этого массива элементы.

damalikov commented on src/lesson4/Task6/Person.java:
```diff
@@ -0,0 +1,97 @@
+        System.out.println(person[index]);
+    }
+
+    public static Person[] returnStudents(Person[] person){
```
Ну явно в задании не было, но вообще надо было просто сделать еще один класс. Например University или School или что-нибудь еще. Там хранить массив Person и там реализовать методы. Можешь исправить))

damalikov commented on src/lesson4/Task6/Person.java:
```diff
@@ -0,0 +1,97 @@
+        index = 0;
+        int max = ((Lector)lectors[0]).getWorkingYears();
+        int indexLector = 0;
+        while (lectors[index] != null) {
```
Не понимаю зачем проверять на null. Ты же только что сформровал массив и у тебя их там быть не может. 
Напиши обычный for и найди индекс лектора

damalikov reviewed (approved):
В принципе все не плохо. Я в последней задаче написал комментарии, можешь поправить и смержить.

damalikov commented on src/lesson4/Task1/Numbers.java:
```diff
@@ -0,0 +1,48 @@
+        this.numbers = numbers;
+    }
+
+    public int[] getNumbers() {
```
Лучше возвращать копию массива. Напомни если что на занятии я объясню почему. Чтобы всем объяснить.

damalikov commented on src/lesson4/Task1/Numbers.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    private int findMax(){
+        int max = this.numbers[0];
```
если метод main будет в другом классе, то методы с таким модификатором не будут видны. Лучше использвать private

damalikov commented on src/lesson4/Task6/University.java:
```diff
@@ -0,0 +1,86 @@
+
+    public static void oldestMan(Person[] person){
+        int index = 0;
+        int max = person[0].getAge();
```
А что будет , если массив будет пустой?

damalikov commented on src/lesson4/Task6/University.java:
```diff
@@ -0,0 +1,86 @@
+            }
+        }
+        int max = lectors[0].getWorkingYears();
+        int indexLector = 0;
```
А если массив будет пустой? 
Я бы сохранял lectors[0] а не lectors[0].getWorkingYears();
В таком случае у тебя всегда будет сслыка на имеющегося пользователя, не придется хранить дополнительо его index и значение workingYears

Relaxiy committed:
package lesson3;

import java.util.Scanner;

public class Task1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        a[0] = 1;
        a[1] = 1;
        System.out.println(a[0]);
        System.out.println(a[1]);

        for(int i = 2; i < a.length; i++) {
            a[i] = a[i - 1] + a[i - 2];
            System.out.println(a[i]);
        }

    }
}
